### PR TITLE
Warn that members won't be autojoined to upgraded rooms

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/8173

This also adds a confirmation prompt to discourage people from breaking their room when they upgrade using the slash command. Theory is that they would have already received a bunch of warnings and buttons to click if their room really needs upgrading. The command is more or less a development tool anyhow.

*Note to the reviewer*: 3 areas are touched here, as 3 separate commits.

Screenshots of the 3 affected areas (in order of commits):
image

----

image

----

image

### CODE REVIEW

**Comment by turt2live** [comment]
Requesting review for code auditing - wording to be reviewed by @lampholder

File: src/SlashCommands.js [hunk 1]
```diff
@@ -105,7 +106,43 @@ export const CommandMap = {
...
+                    QuestionDialog, {
+                    title: _t('Room upgrade confirmation'),
+                    description: (
+                        <div>
+                            { _t(
+                                "Upgrading your room in this way can be dangerous or unnecessary. Room upgrades " +
+                                "are usually done to change the server's behaviour in a given room and not so much " +
+                                "anything to do with client (Riot) behaviour.",
+                            ) }
+                            <br />
```
Comment (dbkr) [inline src/SlashCommands.js]: These could probably be `<p>`s

**Comment by lampholder** [comment]
What about some words like:

-----
This room is running room version `1`, which this homeserver has marked as _unstable_.

Rooms running unstable room versions might have bugs, missing features or security vulnerabilities - we recommend that you upgrade this room to a stable room version.

Upgrading this room will shut down the current instance of the room and create an upgraded room with the same name.

**Warning**: Upgrading a room will _not automatically migrate room members to the new version of the room_. We'll post a link to the new room in the old version of the room - room members will have to click this link to join the new room.

[ Upgrade this room to the recommended room version ]

Only room administrators will see this warning.

-----

### Room version
Room version: state-v2-test

**Warning**: Upgrading a room will _not automatically migrate room members to the new version of the room_. We'll post a link to the new room in the old version of the room - room members will have to click this link to join the new room.

[ Upgrade this room to the recommended room version ]

-----

## Room upgrade confirmation

Upgrading a room can be destructive and isn't always necessary.

Room upgrades are usually recommended when a room version is considered _unstable_. Unstable room versions might have bugs, missing features or security vulnerabilities.

Room upgrades usually only affect _server-side_ processing of the room. If you're having problems with your Riot client please file a an issue with https://github.com/vector-im/riot-web/issue.

**Warning**: Upgrading a room will _not automatically migrate room members to the new version of the room_. We'll post a link to the new room in the old version of the room - room members will have to click this link to join the new room.

Please confirm that you'd like to upgrade this room from `state-v2-test` to `3`.

[ Cancel ] [ Upgrade ]

----------

**Comment by turt2live** [comment]
The translation updates are a bit more complex than I expected - @dbkr please take a quick look at the new changes. Might be easier to just start over, sorry :(

**Comment by dbkr** [comment]
re-lgtm

**Pull request merged**